src/proxyNetworkProvider.ts

```diff
@@ -99,15 +99,29 @@ export class ProxyNetworkProvider implements INetworkProvider {
         return tokenData;
     }
 
-    async getTransaction(txHash: string): Promise<TransactionOnNetwork> {
+    async getTransaction(txHash: string, withProcessStatus?: boolean): Promise<TransactionOnNetwork> {
+        let processStatusPromise: Promise<TransactionStatus> | undefined;
+
+        if (withProcessStatus === true) {
+            processStatusPromise = this.getTransactionStatus(txHash);
+        }
+
         let url = this.buildUrlWithQueryParameters(`transaction/${txHash}`, { withResults: "true" });
         let response = await this.doGetGeneric(url);
-        let transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
+
+        let transaction: TransactionOnNetwork;
```

Comment:
You can directly return the transaction. This code 
```
let transaction: TransactionOnNetwork;
if (processStatusPromise !== undefined) {
    const processStatus = await processStatusPromise;
    transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction, processStatus);
}
else {
    transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
}
```

could be refactored like this:

```
if (processStatusPromise) {
const processStatus = await processStatusPromise;
   return TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction, processStatus);
}
return TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
```

Reply:
Indeed! Done! Thanks for the suggestion.